src/librbd/io/ImageRequestWQ.cc

```diff
@@ -557,6 +588,14 @@ void *ImageRequestWQ<I>::_void_dequeue() {
     ThreadPool::PointerWQ<ImageRequest<I> >::_void_dequeue());
   assert(peek_item == item);
 
+  if (item->needs_throttle() && iops_throttle->get(1, new C_Throttle(this, item))) {
```

Comment:
Memory leak of ```C_Throttle``` if not throttled. I would still like to see the ```get``` method have a template wrapper that takes a callback function, ```this``` pointer, and payload pointer similar to ```ObjectMap::aio_update``` so we don't do an unnecessary extra memory allocation per IO when not throttled.

Reply:
Okey, then I will pass  a ```this``` pointer to get.

Reply:
@dillaman  What about just make ImageRequestWQ extend the Context. then we can pass this to get(). Or add a template for get() and make ImageRequestWQ to extent a new class maybe "ImageRequestWQHandle" ?

Reply:
Not sure I am following